src/analysis/processing/qgsoverlayutils.cpp

```diff
@@ -73,189 +73,142 @@ static bool sanitizeDifferenceResult( QgsGeometry &geom )
   return true;
 }
 
-
-void QgsOverlayUtils::difference( const QgsFeatureSource &sourceA, const QgsFeatureSource &sourceB, QgsFeatureSink &sink, QgsProcessingContext &context, QgsProcessingFeedback *feedback, int &count, int totalCount, QgsOverlayUtils::DifferenceOutput outputAttrs )
+QgsFeatureList QgsOverlayUtils::featureIntersection( const QgsFeature &featA, const QgsFeatureSource &sourceA, const QgsFeatureSource &sourceB, QgsSpatialIndex &indexB, const QgsCoordinateTransformContext &transformContext, const QList<int> &fieldIndicesA, const QList<int> &fieldIndicesB )
```

Comment:
can you please swap back the order of these two functions? Hopefully it will make the diff cleaner and easier to review. As it stands it's basically impossible to reliably review this change.